lib/MadNLPTests/src/Instances/hs15.jl

```diff
@@ -1,9 +1,9 @@
-struct HS15Model <: NLPModels.AbstractNLPModel{Float64,Vector{Float64}}
-    meta::NLPModels.NLPModelMeta{Float64, Vector{Float64}}
+struct HS15Model{T} <: NLPModels.AbstractNLPModel{T,Vector{T}}
+    meta::NLPModels.NLPModelMeta{T, Vector{T}}
     counters::NLPModels.Counters
 end
 
-function HS15Model(; x0=zeros(2), y0=zeros(2))
+function HS15Model(T = Float64; x0=zeros(T,2), y0=zeros(T,2))
```

Comment:
I would maybe pass `T` directly in the optional arguments after the semicolon `;`?